### PR TITLE
Introduce AirbyteConfig interface

### LINKED ISSUE #4880: Refactor ConfigSchema enum for flexibility
## Tell us about the problem you're trying to solve
We need to refactor the `ConfigPersistence` to make it more flexible and reduce code duplication on config persistence.

## Describe the solution you’d like
`ConfigPersistence` interface & class hierarchy should handle `string` config types instead of enum so that if some enum value is removed from the enum in the future, the code is better able to manage it.

It is also more easily to add new config types.

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Closes https://github.com/airbytehq/airbyte/issues/4880

## Recommended reading order
1. `x.java`
2. `y.python`

### CODE REVIEW

**Review by tuliren — APPROVED** [review]
Can you add a test case for the `deleteConfig` method in `DatabaseConfigPersistenceTest`?

File: airbyte-config/models/src/main/java/io/airbyte/config/AbstractConfig.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.config;
+
+import java.io.File;
+
+public interface AbstractConfig {
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/AbstractConfig.java:29]: Spoke offline: Comment here to explain what this interface is for.

File: airbyte-config/models/src/main/java/io/airbyte/config/AbstractConfig.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  String name();
+
+  String getIdFieldName();
+
+  <T> String getId(T config);
+
+  File getFile();
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/AbstractConfig.java:37]: spoke offline: this method can be better named.

**Review by davinchia — APPROVED** [review]
Nice! I think we can make the interface clearer with better naming + comments.

**Pull request merged**